src/System.Memory/src/System/Buffers/BuffersExtensions.cs

```diff
@@ -115,6 +115,9 @@ public static void Write<T>(this IBufferWriter<T> writer, ReadOnlySpan<T> value)
         {
             Span<T> destination = writer.GetSpan();
 
+            if (destination.IsEmpty)
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.writer);
```

Comment:
Is this check actually necessary?  Seems we'd avoid the infinite loop without it.  It's fine to have it, but since writer.GetSpan returning an empty span is a violation of the contract, seems reasonable to omit this if there's any measurable benefit to doing so.

Reply:
You are right, it was not necessary. It turned out that if we want to be consistent with copying an empty Span to an empty Span the Writer should not throw for such case.

I added tests to cover both single-segment and multiple segments cases.

```cs
Span<byte>.Empty.CopyTo(Span<byte>.Empty);
```